pkg/deploy/kubernetes_image_puller.go

```diff
@@ -44,6 +44,14 @@ type ImageAndName struct {
 // a KubernetesImagePuller CR.  Add a finalizer to the CheCluster CR.  If false, remove the KubernetesImagePuller CR, uninstall the operator, and remove the finalizer.
 func ReconcileImagePuller(ctx *DeployContext) (reconcile.Result, error) {
 
+	if !ctx.CheCluster.ObjectMeta.DeletionTimestamp.IsZero() {
+		reconcileResult, err := DeelteImagePullerOperatorAndFinalizer(ctx)
```

Comment:
It is better not to allow return nil.
Then we can do simply `return DeleteImagePullerOperatorAndFinalizer(ctx)`